### PR TITLE
Fix basetoken related issues on in publish and asset detail pages

### LINKED ISSUE #1855: Basetoken address network mismatch with the asset's network causes the price spinner to indefinitely load
Example: https://market.oceanprotocol.com/asset/did:op:2f2cd3221dfc80540d753cfad6859fe84e62327f5d7f95387170f4f4136b7d82
The asset is published on polygon but basetoken address is incorrectly pointing to the eth-mainnet.
The behavior of the market is confusing as it doesn't display an error but keeps indefinitely loading the price.

@bogdanfazakas offered a great suggestion: To display the message that the pricing schema is not available.

### LINKED ISSUE #1863: BaseToken not updated correctly on network change
if the network is switched on any step other than pricing step it doesn’t switch the baseToken address for corresponding network address, only if go through that pricing step after the network switch it updates to the corect baseToken,

### PR DESCRIPTION
**Pull request by bogdanfazakas**
- Fixes #1855 
- Fixes #1863

Changes proposed in this PR:

- fix infinite price loading
   test asset: https://market.oceanprotocol.com/asset/did:op:2f2cd3221dfc80540d753cfad6859fe84e62327f5d7f95387170f4f4136b7d82
- fixed baseToken update in publish form when switching networks
- fix footer text when there are ve Allocations but no sales or sales are not available, see issue below
Screenshot 2023-01-16 at 15 42 03

### CODE REVIEW

**Review by EnzoVezzaro — APPROVED** [review]
LGTM

File: src/components/Asset/AssetActions/Download.tsx [hunk 1]
```diff
@@ -50,11 +50,16 @@ export default function Download({
     useState<OrderPriceAndFees>()
   const [retry, setRetry] = useState<boolean>(false)
 
-  const isUnsupportedPricing = asset?.accessDetails?.type === 'NOT_SUPPORTED'
+  console.log(asset?.accessDetails)
```
Comment (mihaisc) [inline src/components/Asset/AssetActions/Download.tsx:53]: remove console

**Comment by mihaisc** [comment]
So i filled the from on goerli until the last step, then i switched to mumbai and i got this error at `Create token and pricing` 
image

**Comment by mihaisc** [comment]
another test, didn't get an error, just stuck in loop, switched from mumbai to goerli in last step. The basetoken address is updated .
image

**Comment by EnzoVezzaro** [comment]
> another test, didn't get an error, just stuck in loop, switched from mumbai to goerli in last step. The basetoken address is updated .

I couldn't replicate this. I can change to any network in the last step without issues (basetoken gets updated, tx runs correctly).

**Review by kremalicious — APPROVED** [review]

**Pull request merged**